[wasm][debugger] Avoid infinite loop when we have a boxed `new object`

radical opened the pull request:
Eg. `object o = new object(); object o1 = o;`
- Avoid infinitely looping for `o1`

Fixes https://github.com/dotnet/runtime/issues/41990

radical committed:
[wasm][debugger] Avoid infinite loop when we have a boxed `new object`

Eg. `object o = new object(); object o1 = o;`
- Avoid infinitely looping for `o1`

lambdageek commented:
I'm not super familiar with this part of the codebase, but what happens if I do
```csharp
  ValueType v = new S ();
```
where `S` is some struct?

radical committed:
[wasm][debugger] Handle valuetypes boxed in classes that are not

.. type `object`.

Prompted by @lambdageek's comment - https://github.com/dotnet/runtime/pull/42059#issuecomment-690310274

radical commented:
> I'm not super familiar with this part of the codebase, but what happens if I do
> 
> ```cs
>   ValueType v = new S ();
> ```
> 
> where `S` is some struct?

This should fix it - https://github.com/dotnet/runtime/pull/42059/commits/35f70e04c74879ca5112d528f15eaa7dcaf427a2 .
